test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
`test-integration` sets up these variables already [here](https://github.com/openshift/origin/blob/master/hack/test-integration.sh#L9). All temporary files should prefer to use `$BASETMPDIR` or `$ARTIFACT_DIR` for temporary files and only default to a system temp-dir if these variables are not set. You are solving the same problem.